daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/Preprocessor.scala

```diff
@@ -86,18 +86,23 @@ private[engine] final class Preprocessor(
         ResultDone(acc.toList)
     }
 
+  @tailrec
```

Comment:
This change seems unrelated to the `byInterface` field. I'm not sure if there are any performance implications for either but I prefer the version with `foldLeft` since it's shorter and avoids manual recursion.

Reply:
You are right.

Reply:
A revert PR is by essence strongly conservative; I would prefer to set aside any handwritten changes for a separate PR and only accept `git`'s changes for such a thing.

The `foldLeft` version was an improvement, but kind of by definition we're "undoing improvements" 🥲

Reply:
I cherry pick only what I need in the PR from #13491. 
You will add to review against. sorry.

Reply:
that's okay, cherry-picking is the best we can do given that we enforce squash-merges